### PR TITLE
[FIX] hr_expense: Fix payment method line flow

### PR DESCRIPTION
**Pull request by JulienAlardot**
The payment method line set on the expense sheet isn't the one used
on the payment.

The 'print check' payment method test, requiring
the 'account_check_printing' installation and that module not being
in hr_expense dependencies would fail if both weren't installed.

A new and more method-agnostic test replaces it

Task-3498991

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_expense/tests/test_expenses.py [hunk 1]
```diff
@@ -894,59 +894,6 @@ def test_employee_supplier(self):
             'partner_id': self.expense_user_employee.partner_id.id,
         }])
 
-    def test_print_expense_check(self):
-        """
-        Test the check content when printing a check
-        that comes from an expense
-        """
```
Comment (william-andre) [inline addons/hr_expense/tests/test_expenses.py:901]: Why is this test removed? It seems unrelated to your commit?
And your new test doesn't seem to do the same kind of assertions

File: addons/hr_expense/tests/test_expenses.py [hunk 2]
```diff
@@ -1148,3 +1095,36 @@ def test_expense_from_attachments(self):
         self.env['hr.expense'].create_expense_from_attachments(attachment.id)
         expense = self.env['hr.expense'].search([], order='id desc', limit=1)
         self.assertEqual(expense.account_id, product.property_account_expense_id, "The expense account should be the default one of the product")
+
+    def test_expense_payment_method(self):
+        new_payment_method_line = default_payment_method_line = self.company_data['default_journal_bank'].outbound_payment_method_line_ids[0]
+        check_method = self.env['account.payment.method'].search([('code', '=', 'check_printing'), ('payment_type', '=', 'outbound')], limit=1)
+        if check_method:
```
Comment (william-andre) [inline addons/hr_expense/tests/test_expenses.py:1102]: Why is this dependent on the data of the database?
Test should be independent.

File: addons/hr_expense/tests/test_expenses.py (hunk 1 above)
Comment (JulienAlardot) [reply addons/hr_expense/tests/test_expenses.py:901]: Because it consider `account_check_printing` is installed, which is the case on runbot but hr_expense doesn't depends on it

File: addons/hr_expense/tests/test_expenses.py (hunk 2 above)
Comment (JulienAlardot) [reply addons/hr_expense/tests/test_expenses.py]: I improved it, I was trying to keep some of the old test idea but it was not a good idea indeed.
I need to test when you change the payment method so I'll create a new one instead.

File: addons/hr_expense/tests/test_expenses.py (hunk 1 above)
Comment (william-andre) [reply addons/hr_expense/tests/test_expenses.py:901]: Just do something like this then
```python
if 'check_number' not in self.env['account.payment']._fields:
    self.skipTest('`account_check_printing` is not installed')
```

File: addons/hr_expense/tests/test_expenses.py (hunk 2 above)
Comment (william-andre) [reply addons/hr_expense/tests/test_expenses.py]: test is red now

File: addons/hr_expense/tests/test_expenses.py (hunk 1 above)
Comment (JulienAlardot) [reply addons/hr_expense/tests/test_expenses.py:901]: QDP asked me to remove that change from my refactor (I was directly checking the state of the module though)
And because the payment is what handles that part, another test is taking care of the payment values so no need to check further for these imo
And the new test is handling the rest (when it's green again)

Comment (william-andre) [reply addons/hr_expense/tests/test_expenses.py:901]: > QDP asked me to remove that change from my refactor

Why?

> another test is taking care of the payment values

Which?

Comment (JulienAlardot) [reply addons/hr_expense/tests/test_expenses.py:901]: - https://github.com/odoo/odoo/pull/130244#discussion_r1365408382
- I admit I thought it was more explicit than that but the move from the payment is tested here (the tests are trash since forever, hence their rework in 17.0) https://github.com/odoo/odoo/blob/d8ea48fbefbc8fb1528581d3c3d939f89b7feab9/addons/hr_expense/tests/test_expenses.py#L519-L646

Comment (william-andre) [reply addons/hr_expense/tests/test_expenses.py:901]: It is really not obvious to me where you check that we have a reference to the move from which the payment has been created in that test.
Also, the fact that it has been removed in 17.0 (which I still think is a mistake) doesn't mean it should be removed here?

Comment (william-andre) [reply addons/hr_expense/tests/test_expenses.py:901]: > There is no special behaviour between expense and account_check_printing so I really don't get why we need that test to begin with.

Obviously there is something that didn't work: https://www.odoo.com/web#id=3044141&model=project.task
Something doesn't work: not trivial and might break -> test.
Now of course the test was not optimal as it should have been written in `account_check_printing` by creating a payment for a receipt.
But we don't just remove tests like that, especially when it is completely unrelated. The test still runs when installing only `hr_expense` because the main company is in US and so it installs `account_check_printing` by default.

So either don't touch this or move it where it belongs. But don't delete.

Comment (JulienAlardot) [reply addons/hr_expense/tests/test_expenses.py]: Re-added the test in a slightly edited version
